From 21052b6722134da790635e9257f3e5fe1a0ed856 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Sat, 6 Jul 2024 00:06:45 +0800
Subject: [PATCH 20/25] util: detect LoongArch architecture

Only 64-bit is considered now because 32-bit LoongArch Linux support
doesn't exist in upstream yet.

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 src/util/detect_arch.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/util/detect_arch.h b/src/util/detect_arch.h
index 35385a02b49..413575c2b72 100644
--- a/src/util/detect_arch.h
+++ b/src/util/detect_arch.h
@@ -112,6 +112,14 @@
 #endif
 #endif
 
+#if defined(__loongarch__)
+#if __loongarch_grlen == 64
+#define DETECT_ARCH_LOONGARCH64 1
+#else
+#error "pipe: unknown target loongarch grlen"
+#endif
+#endif
+
 #ifndef DETECT_ARCH_X86
 #define DETECT_ARCH_X86 0
 #endif
@@ -168,4 +176,8 @@
 #define DETECT_ARCH_RISCV64 0
 #endif
 
+#ifndef DETECT_ARCH_LOONGARCH64
+#define DETECT_ARCH_LOONGARCH64 0
+#endif
+
 #endif /* UTIL_DETECT_ARCH_H_ */
-- 
2.45.2

